system/jlib/CMakeLists.txt

```diff
@@ -45,6 +45,8 @@ set (    SRCS
          jexcept.cpp 
          jfile.cpp 
          jflz.cpp 
+         jlz4.cpp
+         jden.cpp
```

Comment:
I see that you exclude the body of the file inside jden.cpp if USE_DENSITY isn't defined.
But it would look cleaner if you conditionally added them to SRCS and INCLUDES here in the cmake file in the 1st place.